viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        movie_dict = {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+    
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+#loop through the dictionaries in the list user_data["watchlist"]
+#if the dictionary key == "title", check if the value == title (parameter)
+#if the value == title, save the dictionary to user_data["watched"]. Then delete the dictionary from user_data["watchlist"]
+
+
+    for dict in user_data["watchlist"]:
+        if dict["title"] == title:
+            user_data["watched"].append(dict)
+            user_data["watchlist"].remove(dict)
+
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+
+    ratings_list = []
+    
+    if len(user_data["watched"]) > 0:
+        for dict in user_data["watched"]:
+            ratings_list.append(dict["rating"])
+        
+        average_rating = sum(ratings_list) / len(ratings_list)
+        
+    else:
+        average_rating = 0.0
+    
+    return average_rating
+    
+    
+def get_most_watched_genre(user_data):
+
+    genre_frequency = {}
+
+
+    if len(user_data["watched"]) > 0:
+        for dict in user_data["watched"]:
+            for k, v, in dict.items():
+                if k == "genre":

```

Comment:
If every movie dictionary has a genre, we can directly access the genre using `dict[genre]`. Imagine we were holding movie objects with considerably more data (director, producers, editors, cast, etc), then we would end up accessing a lot of unnecessary values, especially if we don't exit our loop after we found the value we wanted and performed our operations.

Reply:
I see what you mean! Had to reread this comment. My code is looping over every key in the dictionary to find "genre" whereas we could have accessed the "genre" key directly using dict[genre]. This change would save a lot of time complexity.